supermarq-benchmarks/examples/qre-challenge/Lessons-Learned.md

```diff
@@ -0,0 +1,98 @@
+
+Synthetic qubits, particularly those based on superconducting circuits, are one of the leading platforms in quantum computing. These qubits are realized using Josephson junctions, which allow for the creation of a two-level quantum system with low decoherence times and high controllability. The architecture for superconducting qubits is highly developed, with established techniques for qubit manipulation, readout, and error correction. However, the physical infrastructure required, such as dilution refrigerators, makes scalability and qubit connectivity challenging, especially when dealing with large numbers of qubits.
+
+_Neutral Atom Simulations_
+
+Neutral atom qubits, on the other hand, use individual atoms trapped in optical lattices or tweezers as qubits. These qubits are manipulated using laser pulses to perform quantum gates and entanglement operations. Neutral atom simulations are a promising avenue for large-scale quantum computing due to their natural scalability and the ability to arrange atoms in highly connected, flexible geometries. However, challenges remain in terms of maintaining coherence over long periods and achieving high-fidelity operations across large numbers of qubits.
+
+_Comparison and Implications for QRE_
+
+The choice of qubit architecture—synthetic versus neutral atom—has significant implications for QRE. Superconducting qubits often require careful consideration of connectivity due to their fixed physical layout, while neutral atoms offer more flexibility in qubit arrangement but face challenges in operation fidelity. A comprehensive QRE tool must account for these differences in architecture, adjusting resource estimations based on the type of qubit used, its coherence properties, and the specific operational constraints associated with each platform.
+
+### **Requirements**
+
+_Functional Requirements_
+
+- **Movement Analysis:** The tool must evaluate the impact of qubit movement within the architecture.
+
+- **Atom Reloading:** Consideration of the frequency and impact of atom reloading within quantum operations.
+
+- **Code Teleportation:** Ability to model and assess the implications of code teleportation on qubit resources.
+
+- **Connectivity:** Analysis of qubit connectivity and its effect on the overall architecture.
+
+- **Degree of Parallelization:** Estimation of the impact of parallel operations on qubit usage and efficiency.
+
+- **Custom Assumptions:** Flexibility to incorporate various architectural assumptions and parameters.
+
+_Non-Functional Requirements_
+
+- **Performance:** The tool should provide results in a reasonable timeframe, even for large-scale quantum architectures.
+
+- **Scalability:** Ability to scale with increasing complexity of quantum systems.
+
+- **Usability:** User-friendly interface for easy input of parameters and interpretation of results.
+
+- **Extensibility:** Design should allow for future enhancements and the addition of new features.
```

Comment:
Let's remove this for more specific requirements. For example, we want to be able to do a universal gateset without magic state distillation.  People should be able to switch back and forth between 3D codes, subsystem codes, concatenated codes, and piece-able fault tolerance.